### PR TITLE
Check source capabilities to distinguish touches vs mouse

### LINKED ISSUE #1995: `amp-mode-mouse` class is added to body when clicking play on a amp-video on mobile
able to replicate on nexus 5 (chrome port forwarding) as well as chrome emulation.
tested http://output.jsbin.com/binamezotu (scroll up to last slide which is an amp-video and click play, you will then see an arrow on the right)

on nexus 5 chrome and nexus 5 x(bug occurred on actual device) but didn't seem to be an issue on ios safari and ios chrome (layout was slightly off so might have just been hidden, unfortunately i dont have an iphone to retest).

on everything.amp.html scroll to the amp-video demo, click play. you will amp-mode-mouse added to `body`

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #1995.

### CODE REVIEW

**Comment by cramforce** [comment]
LGTM

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

File: src/input.js [hunk 1]
```diff
@@ -195,8 +195,17 @@ export class Input {
...
+  /**
+   * @param {!Event} e
+   * @return {!Promise|undefined}
+   * @private
+   */
+  onMouseMove_(e) {
+    // The event explicitly states that it's a result of a touch event.
+    if (e.sourceCapabilities && e.sourceCapabilities.firesTouchEvents) {
+      this.mouseCanceled_();
+      return undefined;
```
Comment (sriramkrish85) [inline src/input.js]: remove ?

Comment (dvoytenko) [inline src/input.js]: Remove what, sorry?

Comment (sriramkrish85) [inline src/input.js]: oh i thought the function ended here - 

just return; (would return undefined)

Comment (dvoytenko) [inline src/input.js]: It's fine. Here I'd like to be explicit since below it returns the actual value. Uglifier will likely strip it.

**Pull request merged**